public/javascripts/conversation.js

```diff
@@ -374,6 +375,7 @@ $(() => {
     function appendUserMessage(text) {
         container.append($('<span>').addClass('message message-text from-user')
             .text(text));
+        container.append('<div id="almond_thinking_container" ><div id="almond_thinking_txt"><p>Almond is thinking</p></div><div id="almond_thinking"><span></span><span></span><span></span><span></span><span></span><span></span><span></span></div></div>');
```

Comment:
This doesn't work if you refresh the page: existing messages are resent by the server, and the spinner gets stuck.
Also, hardcoding the message here is problematic for a number of reasons, including translation.
I would do away with the message and entirely and only show the spinner, and also make it fewer dots. 3-5 should be enough.

Reply:
> This doesn't work if you refresh the page: existing messages are resent by the server, and the spinner gets stuck.

Noticed later, trying to figure out a fix for that

> Also, hardcoding the message here is problematic for a number of reasons, including translation.

I know, that is a quick and dirty test, alredy implemented correctly

> I would do away with the message and entirely and only show the spinner, and also make it fewer dots. 3-5 should be enough.

ok